PulseSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java

```diff
@@ -0,0 +1,214 @@
+package com.ooyala.sample.players;
+
+/**
+ * Created by FTT on 30/05/17.
+ */
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.util.Log;
+import android.view.KeyEvent;
+
+import com.facebook.react.modules.core.DefaultHardwareBackBtnHandler;
+import com.ooyala.android.OoyalaNotification;
+import com.ooyala.android.OoyalaPlayer;
+import com.ooyala.android.skin.OoyalaSkinLayoutController;
+import com.ooyala.android.util.SDCardLogcatOoyalaEventsLogger;
+import com.ooyala.sample.utils.VideoItem;
+
+import java.util.Observable;
+import java.util.Observer;
+
+import static android.Manifest.permission.WRITE_EXTERNAL_STORAGE;
+import static android.content.pm.PackageManager.PERMISSION_GRANTED;
+
+/**
+ * This class asks permission for WRITE_EXTERNAL_STORAGE. We need it for automation hooks
+ * as we need to write into the SD card and automation will parse this file.
+ */
+public abstract class AbstractHookActivity extends Activity implements Observer, DefaultHardwareBackBtnHandler {
+	private static final int PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE = 1;
+	String TAG = this.getClass().toString();
+	final int CLICK_THROUGH_REQUEST = 11234;
+	protected OoyalaSkinLayoutController playerSkinLayoutController;
+	final String LOG_TAG = this.getClass().toString();
+
+	SDCardLogcatOoyalaEventsLogger log = new SDCardLogcatOoyalaEventsLogger();
+
+	String embedCode;
+	String pcode;
+	String domain;
+
+	OoyalaPlayer player;
+
+	boolean writePermission = false;
+	boolean asked = false;
+
+	// complete player setup after we asked for permission to write into external storage
+	abstract void completePlayerSetup(final boolean asked);
+
+	@Override
+	protected void onCreate(@Nullable Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+
+		if (ContextCompat.checkSelfPermission(this, WRITE_EXTERNAL_STORAGE) != PERMISSION_GRANTED) {
+			ActivityCompat.requestPermissions(this, new String[]{WRITE_EXTERNAL_STORAGE}, PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE);
+		} else {
+			writePermission= true;
+			asked = true;
+		}
+
+		embedCode = getIntent().getExtras().getString("embed_code");
+		pcode = getIntent().getExtras().getString("pcode");
+		domain = getIntent().getExtras().getString("domain");
+	}
+
+	protected void setPlayerLayoutController(OoyalaSkinLayoutController playerSkinLayoutController) {
```

Comment:
Yes. Now we have defined in parent class in using that in child class as well. removed this method.